backend/src/main/java/com/woowacourse/naepyeon/service/TeamService.java

```diff
@@ -188,9 +195,20 @@ public boolean isJoinedMember(final Long memberId, final Long teamId) {
                 .anyMatch(team -> team.getId().equals(teamId));
     }
 
-    public String createInviteToken(final Long teamId) {
-        validateExistTeam(teamId);
-        return inviteTokenProvider.createInviteToken(teamId);
+    @Transactional
+    public String createInviteCode(final Long teamId) {
+        final Team team = teamRepository.findById(teamId)
+                .orElseThrow(() -> new NotFoundTeamException(teamId));
+        final InviteCode inviteCode = team.createInviteCode(
+                () -> SecureRandomStringUtils.createRandomAlphanumericSecure(10)
+        );
+        try {
+            return inviteCodeRepository.save(inviteCode)
+                    .getCode();
+        } catch (final DataIntegrityViolationException e) {
+            log.debug("초대 코드 중복 키 발생, 재시도");
+            return createInviteCode(teamId);
+        }
```

Comment:
예외 발생 시 재귀적으로 함수를 호출한다면 어차피 200~203번째 라인을 다시할 것 같은데 아래와 같은 형태는 어떤가요?
```suggestion
    @Transactional
    public String createInviteCode(final Long teamId) {
        final Team team = teamRepository.findById(teamId)
                .orElseThrow(() -> new NotFoundTeamException(teamId));
        return makeInviteCode(team);
    }

    private String makeValidInviteCode(final Team team) {
        try {
            final InviteCode inviteCode = team.createInviteCode(
                    () -> SecureRandomStringUtils.createRandomAlphanumericSecure(10)
            );
            return inviteCodeRepository.save(inviteCode)
                    .getCode();
        } catch (final DataIntegrityViolationException e) {
            log.debug("초대 코드 중복 키 발생, 재시도");
            return makeInviteCode(team);
        }
    }
```

Reply:
케이 말대로 findById까지 재귀에 포함시킬 필요는 없는 것 같네요!
메서드 분리 해봤습니당